lib/webconn.js

```diff
@@ -115,100 +117,39 @@ export default class WebConn extends Wire {
       })
     }
 
-    // Now make all the HTTP requests we need in order to load this piece
-    // Usually that's one requests, but sometimes it will be multiple
-    // Send requests in parallel and wait for them all to come back
-    let numRequestsSucceeded = 0
-    let hasError = false
-
-    let ret
-    if (requests.length > 1) {
-      ret = new Uint8Array(length)
-    }
-
-    requests.forEach(request => {
-      const url = request.url
-      const start = request.start
-      const end = request.end
+    const chunks = await Promise.all(requests.map(async ({ start, end, url }) => {
       debug(
         'Requesting url=%s pieceIndex=%d offset=%d length=%d start=%d end=%d',
         url, pieceIndex, offset, length, start, end
       )
-      const opts = {
-        url,
-        method: 'GET',
-        headers: {
-          'user-agent': `WebTorrent/${VERSION} (https://webtorrent.io)`,
-          range: `bytes=${start}-${end}`
-        },
-        timeout: SOCKET_TIMEOUT
+      let res
+      try {
+        res = await fetch(url, {
+          cache: 'no-store',
+          method: 'GET',
+          headers: {
+            'user-agent': `WebTorrent/${VERSION} (https://webtorrent.io)`,
+            range: `bytes=${start}-${end}`
+          },
+          signal: AbortSignal.timeout(SOCKET_TIMEOUT)
```

Comment:
honestly "all requests" will be at most 2-4 requests, this isn't a major concern, and not worth the overhead it would cause for the 1000 requests made